HDFS-15976. Make mkdtemp cross platform

GauthamBanasandra opened the pull request:
* mkdtemp is used for creating temporary
  directory, adhering to the given pattern.
  It's not available on Visual C++. Need
  to make this cross platform.

GauthamBanasandra committed:
HDFS-15976. Make mkdtemp cross platform

* mkdtemp is used for creating temporary
  directory, adhering to the given pattern.
  It's not available on Visual C++. Need
  to make this cross platform.

GauthamBanasandra committed:
Refactor TempDir class

* Refactored TempDir class
  to adhere to the Rule of 5
  C++ idiom.

GauthamBanasandra committed:
Fix documentation

* Fixed the @return section
  in documentation.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 54s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 6 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  38m  4s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   2m 57s |  |  trunk passed with JDK Ubuntu-11.0.10+9-Ubuntu-0ubuntu1.20.04  |
| +1 :green_heart: |  compile  |   2m 52s |  |  trunk passed with JDK Private Build-1.8.0_282-8u282-b08-0ubuntu1~20.04-b08  |
| +1 :green_heart: |  mvnsite  |   0m 22s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  60m  3s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 13s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 42s |  |  the patch passed with JDK Ubuntu-11.0.10+9-Ubuntu-0ubuntu1.20.04  |
| +1 :green_heart: |  cc  |   2m 42s |  |  the patch passed  |
| +1 :green_heart: |  golang  |   2m 42s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 42s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 44s |  |  the patch passed with JDK Private Build-1.8.0_282-8u282-b08-0ubuntu1~20.04-b08  |
| +1 :green_heart: |  cc  |   2m 44s |  |  the patch passed  |
| +1 :green_heart: |  golang  |   2m 44s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 44s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |   0m 15s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  15m 52s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  |  91m 44s | [/patch-unit-hadoop-hdfs-project_hadoop-hdfs-native-client.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/1/artifact/out/patch-unit-hadoop-hdfs-project_hadoop-hdfs-native-client.txt) |  hadoop-hdfs-native-client in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 30s |  |  The patch does not generate ASF License warnings.  |
|  |   | 177m 20s |  |  |


| Reason | Tests |
|-------:|:------|
| Failed CTEST tests | configuration |
|   | memcheck_configuration |
|   | hdfs_configuration |
|   | memcheck_hdfs_configuration |
|   | hdfs_builder_test |
|   | memcheck_hdfs_builder_test |
|   | hdfs_config_connect_bugs |
|   | memcheck_hdfs_config_connect_bugs |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/2908 |
| Optional Tests | dupname asflicense compile cc mvnsite javac unit codespell golang |
| uname | Linux cd68af073374 4.15.0-136-generic #140-Ubuntu SMP Thu Jan 28 05:20:47 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 5df023b90978abf2bf5ec131b5e401c1f880d268 |
| Default Java | Private Build-1.8.0_282-8u282-b08-0ubuntu1~20.04-b08 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.10+9-Ubuntu-0ubuntu1.20.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_282-8u282-b08-0ubuntu1~20.04-b08 |
| CTEST | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/1/artifact/out/patch-hadoop-hdfs-project_hadoop-hdfs-native-client-ctest.txt |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/1/testReport/ |
| Max. process+thread count | 569 (vs. ulimit of 5500) |
| modules | C: hadoop-hdfs-project/hadoop-hdfs-native-client U: hadoop-hdfs-project/hadoop-hdfs-native-client |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/1/console |
| versions | git=2.25.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 54s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 6 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  34m 51s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   2m 38s |  |  trunk passed with JDK Ubuntu-11.0.10+9-Ubuntu-0ubuntu1.20.04  |
| +1 :green_heart: |  compile  |   2m 39s |  |  trunk passed with JDK Private Build-1.8.0_282-8u282-b08-0ubuntu1~20.04-b08  |
| +1 :green_heart: |  mvnsite  |   0m 28s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  54m 12s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 16s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 34s |  |  the patch passed with JDK Ubuntu-11.0.10+9-Ubuntu-0ubuntu1.20.04  |
| +1 :green_heart: |  cc  |   2m 34s |  |  the patch passed  |
| +1 :green_heart: |  golang  |   2m 34s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 34s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 32s |  |  the patch passed with JDK Private Build-1.8.0_282-8u282-b08-0ubuntu1~20.04-b08  |
| +1 :green_heart: |  cc  |   2m 32s |  |  the patch passed  |
| +1 :green_heart: |  golang  |   2m 32s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 32s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |   0m 19s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  13m 21s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  | 116m  2s | [/patch-unit-hadoop-hdfs-project_hadoop-hdfs-native-client.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/2/artifact/out/patch-unit-hadoop-hdfs-project_hadoop-hdfs-native-client.txt) |  hadoop-hdfs-native-client in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 34s |  |  The patch does not generate ASF License warnings.  |
|  |   | 193m  3s |  |  |


| Reason | Tests |
|-------:|:------|
| Failed CTEST tests | configuration |
|   | memcheck_configuration |
|   | hdfs_configuration |
|   | memcheck_hdfs_configuration |
|   | hdfs_builder_test |
|   | memcheck_hdfs_builder_test |
|   | hdfs_config_connect_bugs |
|   | memcheck_hdfs_config_connect_bugs |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/2908 |
| Optional Tests | dupname asflicense compile cc mvnsite javac unit codespell golang |
| uname | Linux 8bfbfed8d78b 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 5df023b90978abf2bf5ec131b5e401c1f880d268 |
| Default Java | Private Build-1.8.0_282-8u282-b08-0ubuntu1~20.04-b08 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.10+9-Ubuntu-0ubuntu1.20.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_282-8u282-b08-0ubuntu1~20.04-b08 |
| CTEST | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/2/artifact/out/patch-hadoop-hdfs-project_hadoop-hdfs-native-client-ctest.txt |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/2/testReport/ |
| Max. process+thread count | 713 (vs. ulimit of 5500) |
| modules | C: hadoop-hdfs-project/hadoop-hdfs-native-client U: hadoop-hdfs-project/hadoop-hdfs-native-client |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/2/console |
| versions | git=2.25.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

GauthamBanasandra committed:
Trigger CI

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |  24m 19s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 6 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  34m 50s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   2m 46s |  |  trunk passed with JDK Ubuntu-11.0.11+9-Ubuntu-0ubuntu2.20.04  |
| +1 :green_heart: |  compile  |   2m 50s |  |  trunk passed with JDK Private Build-1.8.0_292-8u292-b10-0ubuntu1~20.04-b10  |
| +1 :green_heart: |  mvnsite  |   0m 24s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  57m  1s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 15s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 52s |  |  the patch passed with JDK Ubuntu-11.0.11+9-Ubuntu-0ubuntu2.20.04  |
| +1 :green_heart: |  cc  |   2m 52s |  |  the patch passed  |
| +1 :green_heart: |  golang  |   2m 52s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 52s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 45s |  |  the patch passed with JDK Private Build-1.8.0_292-8u292-b10-0ubuntu1~20.04-b10  |
| +1 :green_heart: |  cc  |   2m 45s |  |  the patch passed  |
| +1 :green_heart: |  golang  |   2m 45s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 45s |  |  the patch passed  |
| -1 :x: |  blanks  |   0m  0s | [/blanks-eol.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/3/artifact/out/blanks-eol.txt) |  The patch has 1 line(s) that end in blanks. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply  |
| +1 :green_heart: |  mvnsite  |   0m 18s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  15m 44s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  |  92m 46s | [/patch-unit-hadoop-hdfs-project_hadoop-hdfs-native-client.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/3/artifact/out/patch-unit-hadoop-hdfs-project_hadoop-hdfs-native-client.txt) |  hadoop-hdfs-native-client in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 30s |  |  The patch does not generate ASF License warnings.  |
|  |   | 198m 49s |  |  |


| Reason | Tests |
|-------:|:------|
| Failed CTEST tests | configuration |
|   | memcheck_configuration |
|   | hdfs_configuration |
|   | memcheck_hdfs_configuration |
|   | hdfs_builder_test |
|   | memcheck_hdfs_builder_test |
|   | hdfs_config_connect_bugs |
|   | memcheck_hdfs_config_connect_bugs |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/2908 |
| Optional Tests | dupname asflicense compile cc mvnsite javac unit codespell golang |
| uname | Linux e21dc1fee9ad 4.15.0-136-generic #140-Ubuntu SMP Thu Jan 28 05:20:47 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 5977b9fee13a5f4669029b3ed3492c49ed152b07 |
| Default Java | Private Build-1.8.0_292-8u292-b10-0ubuntu1~20.04-b10 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.11+9-Ubuntu-0ubuntu2.20.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_292-8u292-b10-0ubuntu1~20.04-b10 |
| CTEST | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/3/artifact/out/patch-hadoop-hdfs-project_hadoop-hdfs-native-client-ctest.txt |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/3/testReport/ |
| Max. process+thread count | 517 (vs. ulimit of 5500) |
| modules | C: hadoop-hdfs-project/hadoop-hdfs-native-client U: hadoop-hdfs-project/hadoop-hdfs-native-client |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/3/console |
| versions | git=2.25.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |  42m 48s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 6 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  34m  2s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   2m 34s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 24s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  52m 52s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 14s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 27s |  |  the patch passed  |
| +1 :green_heart: |  cc  |   2m 27s |  |  the patch passed  |
| +1 :green_heart: |  golang  |   2m 27s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 27s |  |  the patch passed  |
| -1 :x: |  blanks  |   0m  0s | [/blanks-eol.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/4/artifact/out/blanks-eol.txt) |  The patch has 1 line(s) that end in blanks. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply  |
| +1 :green_heart: |  mvnsite  |   0m 17s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  15m 39s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  |  91m 43s | [/patch-unit-hadoop-hdfs-project_hadoop-hdfs-native-client.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/4/artifact/out/patch-unit-hadoop-hdfs-project_hadoop-hdfs-native-client.txt) |  hadoop-hdfs-native-client in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 30s |  |  The patch does not generate ASF License warnings.  |
|  |   | 208m 47s |  |  |


| Reason | Tests |
|-------:|:------|
| Failed CTEST tests | x_platform_syscall_test |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/2908 |
| Optional Tests | dupname asflicense compile cc mvnsite javac unit codespell golang |
| uname | Linux 3ec950582c89 4.15.0-147-generic #151-Ubuntu SMP Fri Jun 18 19:21:19 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / cb6f04c8e2a879dc29198c68455e55fd5e8de179 |
| Default Java | Red Hat, Inc.-1.8.0_292-b10 |
| CTEST | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/4/artifact/out/patch-hadoop-hdfs-project_hadoop-hdfs-native-client-ctest.txt |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/4/testReport/ |
| Max. process+thread count | 555 (vs. ulimit of 5500) |
| modules | C: hadoop-hdfs-project/hadoop-hdfs-native-client U: hadoop-hdfs-project/hadoop-hdfs-native-client |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/4/console |
| versions | git=2.9.5 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

GauthamBanasandra committed:
Fix segfault with mkdtemp

* Need to append a null char
  so that mkdtemp can find
  the end of the string. It leds
  to a segfaul otherwise.

GauthamBanasandra committed:
Revert "Trigger CI"

This reverts commit 373af5331724b9fd4689a12341b0afc8d801c4cb.

hadoop-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 59s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 6 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  34m  5s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   2m 33s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 27s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  53m  3s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 15s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 28s |  |  the patch passed  |
| +1 :green_heart: |  cc  |   2m 28s |  |  the patch passed  |
| +1 :green_heart: |  golang  |   2m 28s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 28s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |   0m 15s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  16m 33s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  92m 42s |  |  hadoop-hdfs-native-client in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 30s |  |  The patch does not generate ASF License warnings.  |
|  |   | 169m  5s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/2908 |
| Optional Tests | dupname asflicense compile cc mvnsite javac unit codespell golang |
| uname | Linux 8aeb79593e42 4.15.0-147-generic #151-Ubuntu SMP Fri Jun 18 19:21:19 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 767c52ffc4e56b50583dcd97cec5c84b8026205c |
| Default Java | Red Hat, Inc.-1.8.0_292-b10 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/5/testReport/ |
| Max. process+thread count | 599 (vs. ulimit of 5500) |
| modules | C: hadoop-hdfs-project/hadoop-hdfs-native-client U: hadoop-hdfs-project/hadoop-hdfs-native-client |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/5/console |
| versions | git=2.9.5 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |  25m 52s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 6 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  23m  9s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   2m 47s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 38s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  42m 51s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 20s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 34s |  |  the patch passed  |
| +1 :green_heart: |  cc  |   2m 34s |  |  the patch passed  |
| +1 :green_heart: |  golang  |   2m 34s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 34s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |   0m 21s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  16m 45s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  98m  9s |  |  hadoop-hdfs-native-client in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 38s |  |  The patch does not generate ASF License warnings.  |
|  |   | 190m  2s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/2908 |
| Optional Tests | dupname asflicense compile cc mvnsite javac unit codespell golang |
| uname | Linux ffed998d21e7 4.15.0-147-generic #151-Ubuntu SMP Fri Jun 18 19:21:19 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 767c52ffc4e56b50583dcd97cec5c84b8026205c |
| Default Java | Red Hat, Inc.-1.8.0_302-b08 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/5/testReport/ |
| Max. process+thread count | 517 (vs. ulimit of 5500) |
| modules | C: hadoop-hdfs-project/hadoop-hdfs-native-client U: hadoop-hdfs-project/hadoop-hdfs-native-client |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/5/console |
| versions | git=2.27.0 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |  15m  8s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 6 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  29m 22s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   2m 28s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 36s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  53m  0s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 14s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 19s |  |  the patch passed  |
| +1 :green_heart: |  cc  |   2m 19s |  |  the patch passed  |
| +1 :green_heart: |  golang  |   2m 19s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 19s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |   0m 15s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  20m 15s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  99m 45s |  |  hadoop-hdfs-native-client in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 29s |  |  The patch does not generate ASF License warnings.  |
|  |   | 193m 52s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/2908 |
| Optional Tests | dupname asflicense compile cc mvnsite javac unit codespell golang |
| uname | Linux ec62cab44c32 4.15.0-147-generic #151-Ubuntu SMP Fri Jun 18 19:21:19 UTC 2021 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 767c52ffc4e56b50583dcd97cec5c84b8026205c |
| Default Java | Debian-11.0.12+7-post-Debian-2 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/5/testReport/ |
| Max. process+thread count | 557 (vs. ulimit of 5500) |
| modules | C: hadoop-hdfs-project/hadoop-hdfs-native-client U: hadoop-hdfs-project/hadoop-hdfs-native-client |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/5/console |
| versions | git=2.30.2 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

GauthamBanasandra committed:
Fix segfault in temp file and dir

* Appending null to the temp
  file name so that _mktemp_s
  can find the end of string and
  doesn't run into segfault.

hadoop-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   1m  0s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 6 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  22m 43s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   2m 56s |  |  trunk passed with JDK Ubuntu-11.0.11+9-Ubuntu-0ubuntu2.20.04  |
| +1 :green_heart: |  compile  |   2m 54s |  |  trunk passed with JDK Private Build-1.8.0_292-8u292-b10-0ubuntu1~20.04-b10  |
| +1 :green_heart: |  mvnsite  |   0m 25s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  45m 10s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 14s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 43s |  |  the patch passed with JDK Ubuntu-11.0.11+9-Ubuntu-0ubuntu2.20.04  |
| +1 :green_heart: |  cc  |   2m 43s |  |  the patch passed  |
| +1 :green_heart: |  golang  |   2m 43s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 43s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 45s |  |  the patch passed with JDK Private Build-1.8.0_292-8u292-b10-0ubuntu1~20.04-b10  |
| +1 :green_heart: |  cc  |   2m 45s |  |  the patch passed  |
| +1 :green_heart: |  golang  |   2m 45s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 45s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |   0m 16s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  15m 44s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 101m 54s |  |  hadoop-hdfs-native-client in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 29s |  |  The patch does not generate ASF License warnings.  |
|  |   | 172m 38s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/2908 |
| Optional Tests | dupname asflicense compile cc mvnsite javac unit codespell golang |
| uname | Linux 4d84517392bd 4.15.0-147-generic #151-Ubuntu SMP Fri Jun 18 19:21:19 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 767c52ffc4e56b50583dcd97cec5c84b8026205c |
| Default Java | Private Build-1.8.0_292-8u292-b10-0ubuntu1~20.04-b10 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.11+9-Ubuntu-0ubuntu2.20.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_292-8u292-b10-0ubuntu1~20.04-b10 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/5/testReport/ |
| Max. process+thread count | 518 (vs. ulimit of 5500) |
| modules | C: hadoop-hdfs-project/hadoop-hdfs-native-client U: hadoop-hdfs-project/hadoop-hdfs-native-client |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/5/console |
| versions | git=2.25.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |  42m 46s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 6 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  33m 42s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   2m 33s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 23s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  52m 48s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 14s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 25s |  |  the patch passed  |
| +1 :green_heart: |  cc  |   2m 25s |  |  the patch passed  |
| +1 :green_heart: |  golang  |   2m 25s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 25s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |   0m 16s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  15m 42s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  |  58m 42s | [/patch-unit-hadoop-hdfs-project_hadoop-hdfs-native-client.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/6/artifact/out/patch-unit-hadoop-hdfs-project_hadoop-hdfs-native-client.txt) |  hadoop-hdfs-native-client in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 30s |  |  The patch does not generate ASF License warnings.  |
|  |   | 175m 42s |  |  |


| Reason | Tests |
|-------:|:------|
| Failed CTEST tests | test_libhdfs_threaded_hdfspp_test_shim_static |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/6/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/2908 |
| Optional Tests | dupname asflicense compile cc mvnsite javac unit codespell golang |
| uname | Linux 14e94cab0b61 4.15.0-147-generic #151-Ubuntu SMP Fri Jun 18 19:21:19 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 44521383968097c6bb359193afa1b0fc6936bc63 |
| Default Java | Red Hat, Inc.-1.8.0_292-b10 |
| CTEST | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/6/artifact/out/patch-hadoop-hdfs-project_hadoop-hdfs-native-client-ctest.txt |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/6/testReport/ |
| Max. process+thread count | 519 (vs. ulimit of 5500) |
| modules | C: hadoop-hdfs-project/hadoop-hdfs-native-client U: hadoop-hdfs-project/hadoop-hdfs-native-client |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/6/console |
| versions | git=2.9.5 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

GauthamBanasandra committed:
Trigger CI

hadoop-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   1m  6s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 6 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  34m 15s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   2m 38s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 26s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  53m 20s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 16s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 30s |  |  the patch passed  |
| +1 :green_heart: |  cc  |   2m 30s |  |  the patch passed  |
| +1 :green_heart: |  golang  |   2m 30s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 30s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |   0m 16s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  15m 43s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 100m 22s |  |  hadoop-hdfs-native-client in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 31s |  |  The patch does not generate ASF License warnings.  |
|  |   | 176m 25s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/7/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/2908 |
| Optional Tests | dupname asflicense compile cc mvnsite javac unit codespell golang |
| uname | Linux a34dbe1e1be4 4.15.0-147-generic #151-Ubuntu SMP Fri Jun 18 19:21:19 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 5d21f07592623771e7e1e1a57d9d1b3562a55b19 |
| Default Java | Red Hat, Inc.-1.8.0_292-b10 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/7/testReport/ |
| Max. process+thread count | 528 (vs. ulimit of 5500) |
| modules | C: hadoop-hdfs-project/hadoop-hdfs-native-client U: hadoop-hdfs-project/hadoop-hdfs-native-client |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/7/console |
| versions | git=2.9.5 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 57s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 6 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  22m 58s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   2m 42s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 34s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  42m 18s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 19s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 31s |  |  the patch passed  |
| +1 :green_heart: |  cc  |   2m 31s |  |  the patch passed  |
| +1 :green_heart: |  golang  |   2m 31s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 31s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |   0m 22s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  15m 49s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 113m 14s |  |  hadoop-hdfs-native-client in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 40s |  |  The patch does not generate ASF License warnings.  |
|  |   | 178m 40s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/7/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/2908 |
| Optional Tests | dupname asflicense compile cc mvnsite javac unit codespell golang |
| uname | Linux 235fc3c80e65 4.15.0-147-generic #151-Ubuntu SMP Fri Jun 18 19:21:19 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 5d21f07592623771e7e1e1a57d9d1b3562a55b19 |
| Default Java | Red Hat, Inc.-1.8.0_302-b08 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/7/testReport/ |
| Max. process+thread count | 579 (vs. ulimit of 5500) |
| modules | C: hadoop-hdfs-project/hadoop-hdfs-native-client U: hadoop-hdfs-project/hadoop-hdfs-native-client |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/7/console |
| versions | git=2.27.0 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   1m  2s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 6 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  29m 53s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   2m 29s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 22s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  53m  9s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 14s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 20s |  |  the patch passed  |
| +1 :green_heart: |  cc  |   2m 20s |  |  the patch passed  |
| +1 :green_heart: |  golang  |   2m 20s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 20s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |   0m 15s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  20m  7s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 103m 29s |  |  hadoop-hdfs-native-client in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 27s |  |  The patch does not generate ASF License warnings.  |
|  |   | 183m 32s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/7/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/2908 |
| Optional Tests | dupname asflicense compile cc mvnsite javac unit codespell golang |
| uname | Linux e9bf7e8bc13e 4.15.0-147-generic #151-Ubuntu SMP Fri Jun 18 19:21:19 UTC 2021 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 5d21f07592623771e7e1e1a57d9d1b3562a55b19 |
| Default Java | Debian-11.0.12+7-post-Debian-2 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/7/testReport/ |
| Max. process+thread count | 607 (vs. ulimit of 5500) |
| modules | C: hadoop-hdfs-project/hadoop-hdfs-native-client U: hadoop-hdfs-project/hadoop-hdfs-native-client |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/7/console |
| versions | git=2.30.2 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 59s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 6 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  22m 31s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   2m 51s |  |  trunk passed with JDK Ubuntu-11.0.11+9-Ubuntu-0ubuntu2.20.04  |
| +1 :green_heart: |  compile  |   3m  8s |  |  trunk passed with JDK Private Build-1.8.0_292-8u292-b10-0ubuntu1~20.04-b10  |
| +1 :green_heart: |  mvnsite  |   0m 24s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  45m  3s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 15s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 43s |  |  the patch passed with JDK Ubuntu-11.0.11+9-Ubuntu-0ubuntu2.20.04  |
| +1 :green_heart: |  cc  |   2m 43s |  |  the patch passed  |
| +1 :green_heart: |  golang  |   2m 43s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 43s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 45s |  |  the patch passed with JDK Private Build-1.8.0_292-8u292-b10-0ubuntu1~20.04-b10  |
| +1 :green_heart: |  cc  |   2m 45s |  |  the patch passed  |
| +1 :green_heart: |  golang  |   2m 45s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 45s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |   0m 15s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  15m 34s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 109m 42s |  |  hadoop-hdfs-native-client in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 29s |  |  The patch does not generate ASF License warnings.  |
|  |   | 180m  6s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/7/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/2908 |
| Optional Tests | dupname asflicense compile cc mvnsite javac unit codespell golang |
| uname | Linux da00c8220415 4.15.0-147-generic #151-Ubuntu SMP Fri Jun 18 19:21:19 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 5d21f07592623771e7e1e1a57d9d1b3562a55b19 |
| Default Java | Private Build-1.8.0_292-8u292-b10-0ubuntu1~20.04-b10 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.11+9-Ubuntu-0ubuntu2.20.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_292-8u292-b10-0ubuntu1~20.04-b10 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/7/testReport/ |
| Max. process+thread count | 593 (vs. ulimit of 5500) |
| modules | C: hadoop-hdfs-project/hadoop-hdfs-native-client U: hadoop-hdfs-project/hadoop-hdfs-native-client |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2908/7/console |
| versions | git=2.25.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.
